### ISSUE TITLE
Issue on htmlblock render

### ISSUE THREAD

**Issue by uazo** [aspnet/Blazor#1198]
Consider this (please note: all in 1 line without space)
```
<NavLink class="nav-link" href="fetchdata"><span class="oi oi-list-rich" aria-hidden="true"></span> Fetch data</NavLink></li>
```

extension will compile with

```
builder.OpenComponent<Microsoft.AspNetCore.Blazor.Routing.NavLink>(48);
builder.AddAttribute(49, "class", "nav-link");
builder.AddAttribute(50, "href", "fetchdata");
builder.AddMarkupContent(51, "<span class=\"oi oi-list-rich\" aria-hidden=\"true\"></span>");
builder.AddAttribute(52, "ChildContent", (Microsoft.AspNetCore.Blazor.RenderFragment)((builder2) => {
    builder2.AddContent(53, " Fetch data");
}
));
builder.CloseComponent();
```

that's incorrect because add an attribute after a content.

**Comment by SteveSandersonMS** [comment]
Fixed with 8c452ff71ad1c59af0115747794a313d91ace789

We'll apply a more long-term fix when we reintroduce the markup optimization later.

**Comment by rynowak** [comment]
I've merged the PR #1209 that addresses the root cause of this
